app/controllers/posts_controller.rb

```diff
@@ -12,6 +12,8 @@ def index
   # GET /posts/1.json
   def show
     @posts=Post.find(params[:id])
+    @check=@posts.likes.exists?(:user_id => current_user[:id])
```

Comment:
Make a new method in likes method in Likes Model and use it views.
Controller is not place for this code.

Reply:
I have already done this but I forgot to remove that line.

Reply:
As I have mention this so many times. 
Clean your code before publishing your work. 
Please take responsibility  of your work.